### PR TITLE
MEGAHIT: Add support for multi-libraries, export log

### PR DESCRIPTION
**Pull request by jfy133**
This adds support for multiply libraries (coassembly?) to MEGAHIT.

I had to separate `reads` into `reads1` `reads2` because I found if it was too nested, then `path` would give an error that it 'Array' is not a path object.

I also added a `log` file export, as this is supported by MultiQC. 

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Review by Joon-Klaps — APPROVED** [review]
Thanks for the quick fix! 🎉

**Review by Joon-Klaps — CHANGES_REQUESTED** [review]
Minor layout changes

File: modules/nf-core/megahit/main.nf [hunk 1]
```diff
@@ -1,22 +1,20 @@
 process MEGAHIT {
-    tag "$meta.id"
+    tag "${meta.id}"
     label 'process_high'
-
     conda "${moduleDir}/environment.yml"
-    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'oras://community.wave.seqera.io/library/megahit_pigz:657d77006ae5f222' :
-        'community.wave.seqera.io/library/megahit_pigz:87a590163e594224' }"
+    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ? 'oras://community.wave.seqera.io/library/megahit_pigz:657d77006ae5f222' : 'community.wave.seqera.io/library/megahit_pigz:87a590163e594224'}"
```
[the remark this answers is not in the document]
Comment (jfy133) [reply modules/nf-core/megahit/main.nf:5]: Ugh this and the other are both weird formatting things

[the remark this answers is not in the document]
Comment (jfy133) [reply modules/nf-core/megahit/main.nf:5]: (experimenting with a new formatter)

**Comment by jfy133** [comment]
Done, sorry abotu that @Joon-Klaps !

**Review by Joon-Klaps — APPROVED** [review]

**Pull request merged**